addons/mrp/models/stock_rule.py

```diff
@@ -66,6 +81,23 @@ def _get_custom_move_fields(self):
         fields += ['bom_line_id']
         return fields
 
+    def _get_domain_run_manufacture(self, bom, procurement):
+        domain = [
+            ('state', '=', 'confirmed'),
+            ('product_id', '=', procurement.product_id.id),
+            ('bom_id', '=', bom.id),
+            ('picking_type_id', '=', self.picking_type_id.id or procurement.values['warehouse_id'].manu_type_id.id),
+            ('company_id', '=', procurement.values.get('company_id', self.env.company).id),
+            ('orderpoint_id', '!=', False),
+            ('orderpoint_id', '=', 'orderpoint_id' in procurement.values and procurement.values['orderpoint_id'].id),
```

Comment:
if orderpoint_id is not in procurement you have ('orderpoint_id', '!=', False) and ('orderpoint_id', '=', False)

Reply:
> also the commit message indicates only mo created by a RR will be merged but it's not true i think? all mo created by run will be affected?

That's the point